kratos/solving_strategies/strategies/line_search_strategy.h

```diff
@@ -274,56 +340,99 @@ class LineSearchStrategy
         typename TSchemeType::Pointer pScheme = this->GetScheme();
         typename TBuilderAndSolverType::Pointer pBuilderAndSolver = this->GetBuilderAndSolver();
 
-        TSystemVectorType aux(b.size()); //TODO: do it by using the space
-        TSparseSpace::Assign(aux,0.5, Dx);
-
-        //compute residual without update
+        TSystemVectorType aux(TSparseSpace::Size(b));
+        
+        double x1 = mFirstAlphaValue;
+        double x2 = mSecondAlphaValue;
+
+        bool converged = false;
+        int it = 0;
+        double xprevious = 0.0;
+
+        //compute residual with 1 coefficient update (x1)
+        //since no update was performed yet, this includes an increment wrt the previous
+        //solution of x1*Dx
+        TSparseSpace::Assign(aux,x1-xprevious, Dx);
+        xprevious = x1;
+        BaseType::UpdateDatabase(A,aux,b,MoveMesh); 
         TSparseSpace::SetToZero(b);
         pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double ro = TSparseSpace::TwoNorm(b);
-
-        //compute half step residual
-        BaseType::UpdateDatabase(A,aux,b,MoveMesh);
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double rh = TSparseSpace::TwoNorm(b);
-
-        //compute full step residual (add another half Dx to the previous half)
-        BaseType::UpdateDatabase(A,aux,b,MoveMesh);
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double rf = TSparseSpace::TwoNorm(b);
-
-        //compute optimal (limited to the range 0-1)
-        //parabola is y = a*x^2 + b*x + c -> min/max for
-        //x=0   --> r=ro
-        //x=1/2 --> r=rh
-        //x=1   --> r =
-        //c= ro,     b= 4*rh -rf -3*ro,  a= 2*rf - 4*rh + 2*ro
-        //max found if a>0 at the position  xmax = (rf/4 - rh)/(rf - 2*rh);
-        double parabola_a = 2*rf + 2*ro - 4*rh;
-        double parabola_b = 4*rh - rf - 3*ro;
-        double xmin = 1e-3;
-        double xmax = 1.0;
-        if( parabola_a > 0) //if parabola has a local minima
+        double r1 = TSparseSpace::Dot(aux,b);
+        
+        double rmax = std::abs(r1);
+        while(!converged && it < mMaxLineSearchIterations)
         {
-            xmax = -0.5 * parabola_b/parabola_a; // -b / 2a
-            if( xmax > 1.0)
-                xmax = 1.0;
-            else if(xmax < -1.0)
-                xmax = -1.0;
-        }
-        else //parabola degenerates to either a line or to have a local max. best solution on either extreme
-        {
-            if(rf < ro)
-                xmax = 1.0;
+
+            //compute residual with 2 coefficient update (x2)
```

Comment:
Comments should start by capital, looks more pro

Reply:
done